src/pages/en/core-concepts/sharing-state.md

```diff
@@ -0,0 +1,685 @@
+```
+</Fragment>
+<Fragment slot="vue">
+```vue
+<!--src/components/CartFlyoutToggle.vue-->
+<template>
+  <!--write to the imported store using `.set`-->
+  <button @click="isCartOpen.set(!$isCartOpen)">Cart</button>
+</template>
+
+<script setup>
+  import { isCartOpen } from '../cartStore';
+  import { useStore } from '@nanostores/vue';
+  
+  // read the store value with the `useStore` hook
+  const $isCartOpen = useStore(isCartOpen);
+</script>
+```
+</Fragment>
+</UIFrameworkTabs>
+
+Then, we can read `isCartOpen` from our `CartFlyout` component:
+
+<UIFrameworkTabs>
+<Fragment slot="preact">
+```jsx
+// src/components/CartFlyout.jsx
+import { useStore } from '@nanostores/preact';
+import { isCartOpen } from '../cartStore';
+
+export default function CartFlyout() {
+  const $isCartOpen = useStore(isCartOpen);
+
+  return $isCartOpen ? <aside>...</aside> : null;
+}
+```
+</Fragment>
+<Fragment slot="react">
+```jsx
+// src/components/CartFlyout.jsx
+import { useStore } from '@nanostores/react';
+import { isCartOpen } from '../cartStore';
+
+export default function CartFlyout() {
+  const $isCartOpen = useStore(isCartOpen);
+
+  return $isCartOpen ? <aside>...</aside> : null;
+}
+```
+</Fragment>
+<Fragment slot="solid">
+```jsx
+// src/components/CartFlyout.jsx
+import { useStore } from '@nanostores/solid';
+import { isCartOpen } from '../cartStore';
+
+export default function CartFlyout() {
+  const $isCartOpen = useStore(isCartOpen);
+
+  return $isCartOpen ? <aside>...</aside> : null;
+}
+```
+</Fragment>
+<Fragment slot="svelte">
+```svelte
+<!--src/components/CartFlyout.svelte-->
+<script>
+  import { isCartOpen } from '../cartStore';
+</script>
+
+{#if $isCartOpen}
+<aside>...</aside>
+{/if}
+```
+</Fragment>
+<Fragment slot="vue">
+```vue
+<!--src/components/CartFlyout.vue-->
+<template>
+  <aside v-if="$isCartOpen">...</aside>
+</template>
+
+<script setup>
+  import { isCartOpen } from '../cartStore';
+  import { useStore } from '@nanostores/vue';
+
+  const $isCartOpen = useStore(isCartOpen);
+</script>
+```
+</Fragment>
+</UIFrameworkTabs>
+
+### Using nanostore "maps"
+
+:::tip
+**[Nanostore `map`s](https://github.com/nanostores/nanostores#maps) are a great choice for objects you write to regularly!** Alongside the standard `get()` and `set()` helpers an `atom` provides, you'll also have a `.setKey()` function to efficiently update individual object keys.
+:::
+
+Now, let's keep track of the items inside your cart. To avoid duplicates and keep track of "quantity," we may store your cart as an object with the item's ID as a key. We'll use a [nanostore `map`](https://github.com/nanostores/nanostores#maps) for this.
+
+Let's add a `cartItem` store to our `cartStore.js` from earlier. You can also switch to a TypeScript file to define the shape if you're so inclined 🙃
+
+
+<Tabs client:visible sharedStore="client-store-example">
+<Fragment slot="tab.js">clientStore.js</Fragment>
+<Fragment slot="tab.ts">clientStore.ts</Fragment>
```

Comment:
Do we have a way to style the tab labels here? Elsewhere in docs we would use backticks to wrap filenames in `<code>` and it might look better here too?

Another option would be to label these “JavaScript” and “TypeScript” and include the filename in the code sample? A standard way to associate filenames with code samples is on my stack in #918 so this could be about to be standardised in any case, so the more generic tab labelling may make more sense in any case once we have that.